### PR TITLE
SLING-11115 : Allow path exemptions for referrer filter

### PR DESCRIPTION
**Pull request by anchela**
@cziegeler , i would appreciate if you had time to review the proposed addition to referrer-filter configuration and the corresponding exclusion during validation.

### CODE REVIEW

File: src/main/java/org/apache/sling/security/impl/ReferrerFilter.java [hunk 1]
```diff
@@ -430,6 +448,22 @@ private boolean isValidRegexReferrer(HostInfo hostInfo) {
...
+    /**
+     * Returns <code>true</code> if the path info associated with the given request is contained in the configured excluded paths.
+     *
+     * @param request The request to check
+     * @return <code>true</code> if the path-info associate with the given request is contained in the configured excluded paths.
+     */
+    private boolean isExcludedPath(HttpServletRequest request) {
+        String path = request.getPathInfo();
+        for (final String excludedPath : this.excludedPaths) {
+            if (excludedPath.equals(path)) {
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/security/impl/ReferrerFilter.java:460]: I would add a null check for excludedPath here, as a sanity check

Comment (anchela) [reply src/main/java/org/apache/sling/security/impl/ReferrerFilter.java:463]: sure -> done

Comment (cziegeler) [reply src/main/java/org/apache/sling/security/impl/ReferrerFilter.java:463]: ah sorry, I meant "excludedPath" :) so a string in the array might be null

Comment (cziegeler) [reply src/main/java/org/apache/sling/security/impl/ReferrerFilter.java:463]: but checking excludedPaths for null is good as well

Comment (anchela) [reply src/main/java/org/apache/sling/security/impl/ReferrerFilter.java:463]: :-)

Comment (anchela) [reply src/main/java/org/apache/sling/security/impl/ReferrerFilter.java]: sodele

Comment (cziegeler) [reply src/main/java/org/apache/sling/security/impl/ReferrerFilter.java]: Thanks

**Pull request merged**